### PR TITLE
Simplify thread naming in TaskStateModelFactory"

### LINKED ISSUE #325: We should use system provided thread Id instead of using a self maintained integer
https://github.com/apache/helix/blob/aa77f5c0bfc3cafb6482b1cef4a1689191cbd846/helix-core/src/main/java/org/apache/helix/task/TaskStateModelFactory.java#L56

Let's do something like this:
thread.setName("TaskStateModelFactory-task_thread" + thread.getId())

### PR DESCRIPTION
**Pull request by desaikomal**
### Issues

- [X] My PR addresses the following Helix issues and references them in the PR description:
Fixes #325 

### Description

- [X] Here are some details about my PR, including screenshots of any UI changes:
In TaskStateModelFactory class, we want to name our thread with a sequence number to uniquely identify threads. The mechanics can be simplified by using the BasicThreadPoolFactory which has way for explicitly naming the thread. 

### Tests

- [ ] The following tests are written for this issue:

(List the names of added unit/integration tests)

- The following is the result of the "mvn test" command on the appropriate module:
mvn -q test run is in progress, will update it as soon as I get the results.

(If CI test fails due to known issue, please specify the issue and test PR locally. Then copy & paste the result of "mvn test" to here.)

### Changes that Break Backward Compatibility (Optional)

- My PR contains changes that break backward compatibility or previous assumptions for certain methods or API. They include:

(Consider including all behavior changes for public methods or API. Also include these changes in merge description so that other developers are aware of these changes. This allows them to make relevant code changes in feature branches accounting for the new method/API behavior.)

### Documentation (Optional)

- In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

**Review by parakhnr — APPROVED** [review]

File: helix-core/src/main/java/org/apache/helix/task/TaskStateModelFactory.java [hunk 1]
```diff
@@ -191,13 +192,7 @@ private static ScheduledExecutorService createThreadPoolExecutor(HelixManager ma
...
         targetThreadPoolSize, manager.getClusterName(), manager.getInstanceName());
-    return Executors.newScheduledThreadPool(targetThreadPoolSize, new ThreadFactory() {
-      private AtomicInteger threadId = new AtomicInteger(0);
-
-      @Override
-      public Thread newThread(Runnable r) {
-        return new Thread(r, "TaskStateModelFactory-task_thread-" + threadId.getAndIncrement());
-      }
-    });
+    return Executors.newScheduledThreadPool(targetThreadPoolSize,
```
Comment (limbooverlambda) [inline helix-core/src/main/java/org/apache/helix/task/TaskStateModelFactory.java:195]: Any reason, we didn't use guava's ThreadFactoryBuilder? https://guava.dev/releases/snapshot/api/docs/com/google/common/util/concurrent/ThreadFactoryBuilder.html#setNameFormat%28java.lang.String%29

**Comment by desaikomal** [comment]
Thanks @parakhnr and @limbooverlambda for review. This PR has been approved by @parakhnr. 
Commit message: "Using BasicThreadFactory to simplify thread naming in TaskStateModelFactory"

**Comment by desaikomal** [comment]
@limbooverlambda - please help review this. thanks

**Review by limbooverlambda — COMMENTED** [review]
LGTM!

**Review by limbooverlambda — APPROVED** [review]

**Comment by desaikomal** [comment]
Thanks @parakhnr and @limbooverlambda for reviewing and approving the change.
Commit message: Simplify thread naming in TaskStateModelFactory.

**Review by parakhnr — APPROVED** [review]

**Pull request merged**